src/Integration.Vsix.UnitTests/Suppression/LiveIssueFactoryTests.cs

```diff
@@ -0,0 +1,280 @@
+        {
+            // Arrange
+            var vsSolutionMock = new Mock<IVsSolution>();
+            uint fileCount = 2; // initialized with a value to actually assign a value to the out
+            vsSolutionMock.Setup(x => x.GetProjectFilesInSolution(0, 0, null, out fileCount))
+                .Returns(VSConstants.S_OK);
+            var fileNames = new string[fileCount];
+            vsSolutionMock.Setup(x => x.GetProjectFilesInSolution(0, fileCount, fileNames, out fileCount))
+                .OutCallback((uint x, uint y, string[] paths, out uint z) =>
+                {
+                    z = 0;
+                    paths[0] = "Project1";
+                    paths[1] = "Project2";
+                });
+            vsSolutionMock.As<IVsSolution5>().Setup(x => x.GetGuidOfProjectFile(It.IsAny<string>()))
+                .Returns(Guid.Empty);
+            var liveIssueFactory = new LiveIssueFactory(new AdhocWorkspace(), vsSolutionMock.Object);
+
+            var diagnostic = Diagnostic.Create(new DiagnosticDescriptor("id", "title", "message", "category",
+                DiagnosticSeverity.Hidden, true), Location.None);
+
+            // Act
+            var result = liveIssueFactory.Create(null, diagnostic);
+
+            // Assert
+            result.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void Create_WhenProjectFilePathIsNull_ReturnsNull()
+        {
+            // Arrange & Act
+            var diagnostic = Diagnostic.Create(new DiagnosticDescriptor("id", "title", "message", "category",
+                DiagnosticSeverity.Hidden, true), Location.None);
+            var result = SetupAndCreate(diagnostic, filePath: null);
+
+            // Assert
+            result.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void Create_WhenProjectFilePathIsNotFoundInDictionary_ReturnsNull()
+        {
+            // Arrange & Act
+            var diagnostic = Diagnostic.Create(new DiagnosticDescriptor("id", "title", "message", "category",
+                DiagnosticSeverity.Hidden, true), Location.None);
+            var result = SetupAndCreate(diagnostic, filePath: "foo");
+
+            // Assert
+            result.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void Create_WhenIssueIsProjectLevel_ReturnsExpectedIssue()
+        {
+            // Arrange & Act
+            var diagnostic = Diagnostic.Create(new DiagnosticDescriptor("id", "title", "message", "category",
+                DiagnosticSeverity.Hidden, true), Location.None);
+            var result = SetupAndCreate(diagnostic, filePath: "Project1");
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Diagnostic.Should().Be(diagnostic);
+            result.ProjectGuid.Should().Be("00000000-0000-0000-0000-000000000000");
+            result.IssueFilePath.Should().Be("");
+            result.StartLine.Should().Be(0);
+            result.WholeLineText.Should().Be("");
```

Comment:
I was wondering about this: would it be better to have _WholeLineText_ == null for project- or file-level issues? That way we could always safely distinguish between project-level issues and other issues that are reported on an empty line (I know you checked we don't have any at the moment).

Reply:
I was tempted to actually have 3 classes: `ProjectLiveIssue`, `FileLiveIssue`, and `LineLiveIssue` or something like this.